src/Repository/StatsContentRepository.php

```diff
@@ -177,106 +121,68 @@ public function getStats(
     {
-        if (null === $end) {
-            $end = new \DateTime();
-        }
-
-        if ($end < $this->start) {
-            throw new \LogicException('End date must be after start date!');
-        }
-
-        $conn = $this->getEntityManager()->getConnection();
-
-        $sql = 'SELECT date_trunc(?, e.created_at) as datetime, count(e.id) as count FROM '.$table.' e
-                    WHERE e.created_at BETWEEN ? AND ?';
-        if ($magazine) {
-            $sql .= ' AND e.magazine_id = ?';
-        }
-        if ($this->onlyLocal) {
-            $sql = $sql.' AND e.ap_id IS NULL';
+        $tableName = match ($tableName) {
+            'entry' => 'entry',
+            'entry_comment' => 'entry_comment',
+            'post' => 'post',
+            'post_comment' => 'post_comment',
+            default => throw new \InvalidArgumentException("$tableName is not a valid countable")
+        };
+
+        $federatedCond = $federated ? ' AND e.ap_id IS NULL ' : '';
+        $magazineCond = $magazine ? 'AND e.magazine_id = :magId' : '';
+        $SinceDateCond = $sinceDate ? 'AND e.created_at > :date' : '';
+        $tilDateCond = $tilDate ? 'AND e.created_at < :untilDate' : '';
+
+        $sql = "SELECT COUNT(e.id) as count FROM $tableName e INNER JOIN public.user u ON e.user_id = u.id WHERE u.is_deleted = false $SinceDateCond $tilDateCond $federatedCond $magazineCond";
+        $rsm = new ResultSetMapping();
+        $rsm->addScalarResult('count', 0);
+        $query = $this->entityManager->createNativeQuery($sql, $rsm);
+
+        if (null !== $sinceDate) {
+            $query->setParameter(':date', $sinceDate);
         }
-        $sql = $sql.' GROUP BY 1 ORDER BY 1';
-
-        $stmt = $conn->prepare($sql);
-        $stmt->bindValue(1, $interval);
-        $stmt->bindValue(2, $this->start, 'datetime');
-        $stmt->bindValue(3, $end, 'datetime');
-        if ($magazine) {
-            $stmt->bindValue(4, $magazine->getId(), ParameterType::INTEGER);
-        }
-
-        return $stmt->executeQuery()->fetchAllAssociative();
-    }
 
-    private function aggregateTotalStats(string $table, ?Magazine $magazine): array
-    {
-        $conn = $this->getEntityManager()->getConnection();
-
-        $sql = 'SELECT COUNT(e.id) as count FROM '.$table.' e';
-        if ($magazine) {
-            $sql .= ' WHERE e.magazine_id = ?';
-        }
-        if ($this->onlyLocal) {
-            $sql = $sql.' AND e.ap_id IS NULL';
+        if (null !== $tilDate) {
+            $query->setParameter(':untilDate', $tilDate);
         }
 
-        $stmt = $conn->prepare($sql);
-        if ($magazine) {
-            $stmt->bindValue(1, $magazine->getId());
+        if (null !== $magazine) {
+            $query->setParameter(':magId', $magazine->getId());
         }
+        $res = $query->getScalarResult();
 
-        return $stmt->executeQuery()->fetchAllAssociative();
+        return $res[0][0];
```

Comment:
check if $res contains an array and length not zero.. And then another `&&` check for the other array (multidimensional array).